api/server/volume.go

```diff
@@ -178,6 +178,13 @@ func (vd *volAPI) create(w http.ResponseWriter, r *http.Request) {
 		vd.sendError(vd.name, method, w, err.Error(), http.StatusInternalServerError)
 		return
 	}
+	spec := dcReq.GetSpec()
+	if spec.VolumeLabels == nil {
+		spec.VolumeLabels = make(map[string]string)
+	}
+	for k, v := range dcReq.Locator.GetVolumeLabels() {
```

Comment:
i'm not sure we need this, it is supposed to be that all spec labels go to the locator instead, right?